instrumentation/log/error_recording_logger.go

```diff
@@ -0,0 +1,116 @@
+package log
+
+import (
+	"fmt"
+	"sync"
+)
+
+type ErrorRecordingLogger struct {
```

Comment:
Are you sure we need another logger? What's the reasoning behind this?

Reply:
not another logger - a decorator